apps/system/js/app_text_selection_dialog.js

```diff
@@ -355,12 +362,21 @@
     return temp;
   };
 
-  AppTextSelectionDialog.prototype._resetCutOrCopiedTimer =
-    function tsd_resetCutOrCopiedTimer() {
-      window.clearTimeout(this._resetCutOrCopiedTimeout);
-      this._resetCutOrCopiedTimeout = window.setTimeout(function() {
-        this._hasCutOrCopied = false;
-      }.bind(this), this.RESET_CUT_OR_PASTE_TIMEOUT);
+  AppTextSelectionDialog.prototype._cancelCutOrCopiedTimer =
+    function tsd_cancelCutOrCopiedTimer() {
+      window.clearTimeout(_cutOrCopiedTimeoutId);
+      _cutOrCopiedTimeoutId = null;
+    };
+
+  AppTextSelectionDialog.prototype._launchCutOrCopiedTimer =
```

Comment:
The timer functions here can be then moved to the GlobalStates instance, since their purpose is only about managing the timeout ID (and the `hasCutOrCopied` state)